compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -1027,6 +1031,17 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       ddef
     }
   }
+
+  private def checkInternalAnnotations(sym: Symbol): Unit = {
+    for {
+      ann <- compilerInternalAnnotations
+      annSym <- sym.getAnnotation(ann)
+      if annSym.pos.isDefined
```

Comment:
Does the typer expand the annotation trees into fully qualified names? If not, you can check for the fully qualified name in `PreTyperComponent` and for the type in `PrepJSInterop` and whitelist fully qualified names.

That solution would not have any false positives.